File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/chain_of_responsibility/equipo_seguridad.dart
import 'handler.dart';
import 'types.dart';

class EquipoSeguridad extends Handler {
  @override
  HandleResult handle(Ticket req, {List<String>? trail}) {
    final t = trail ?? <String>[];
    push(t);
    if (req.category == Category.security) {
      return HandleResult(
        handledBy: 'EquipoSeguridad',
        message: 'Derivado a Seguridad (SLA 1h).',
        trail: t,
      );
    }
    return super.handle(req, trail: t);
  }
}

File: implementations/react-native/design-patterns-rn/src/patterns/behavioral/chain_of_responsibility/equipo_seguridad.ts
import { SupportHandler } from './base_handler';
import type { HandleResult, Ticket } from './types';

export class EquipoSeguridad extends SupportHandler {
  handle(req: Ticket, trail: string[] = []): HandleResult {
    this.push(trail);
    if (req.category === 'security') {
      return {
        handledBy: 'EquipoSeguridad',
        message: 'Derivado a Seguridad (SLA 1h).',
        trail,
      };
    }
    return super.handle(req, trail);
  }
}
